modules/finality-verifier/src/lib.rs

```diff
@@ -142,13 +154,86 @@ pub mod pallet {
 				<Error<T>>::InvalidAncestryProof
 			);
 
-			let _ = T::HeaderChain::append_header(finality_target)?;
+			let _ = T::HeaderChain::append_header(finality_target.clone())?;
```

Comment:
Why do we still need the `HeaderChain`?

Reply:
We don't. I wanted to avoid removing this until later PRs

Reply:
Fair enough. I was actually thinking that you'll start a completely fresh pallet instead of re-using `finality-verifier`, but so be it :)